// Copyright 2013 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.
//
// Distilled page content.

syntax = "proto2";

option optimize_for = LITE_RUNTIME;

package dom_distiller;

message DistilledPageProto {
  // The URLs used to create this page.
  required string url = 1;

  // The distilled HTML.
  required string html = 3;

  message Image {
    // Handle to the image. This could be the URL where the image was found.
    // The handle should be referenced verbatim in the HTML.
    required string name = 1;

    // The image data
    required bytes data = 2;

    // The original url for the image.
    optional string url = 3;
  }

  // The images referenced in the HTML.
  repeated Image image = 4;

  // Title for the current page.
  optional string title = 5;

  message DebugInfo {
    // Contains the log from the JS distiller.
    optional string log = 1;
  }

  // Debug information about this page.
  optional DebugInfo debug_info = 6;

  // text directionality
  optional string text_direction = 7;

  message PaginationInfo {
    optional string next_page = 1;
    optional string prev_page = 2;
    optional string canonical_page = 3;
  }

  // Pagination info for this page.
  optional PaginationInfo pagination_info = 8;

  // Removed obsolete field.
  reserved 9;
}
